lib/active_merchant/billing/gateways/place_to_pay.rb

```diff
@@ -0,0 +1,199 @@
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class PlaceToPayGateway < Gateway
+      self.test_url = 'https://api-co-dev.placetopay.ws/'
+      self.live_url = 'https://api-co.placetopay.ws/'
+
+      self.default_currency = 'COP'
+
+      self.supported_countries = ['COL']
+      self.supported_cardtypes = %i[visa master american_express discover]
+      self.homepage_url = 'https://www.placetopay.com/'
+      self.display_name = 'PlaceToPay'
+
+      STANDARD_ERROR_CODE_MAPPING = {
+        '401' => STANDARD_ERROR_CODE[:processing_error]
+      }
+
+      ISO_FORMAT_DATE = '%Y-%m-%dT%H:%M:%S%:z'
+
+      def initialize(options = {})      
+        requires!(options, :login, :secret_key)
+        @login, @secret_key = options.values_at(:login, :secret_key)
+        super
+      end 
+
+      def purchase(money, payment, options = {})
+        post = {}
+        add_auth(post, options)
+        add_payer(post, options)
+        add_payment(post, options)
+        add_instrument(post, payment, options)
+        commit(:post, 'gateway/process', post)
+      end
+
+      def refund(money, authorization, options = {})
+        post = {}
+        add_auth(post, options)
+        post[:internalReference] = options[:internalReference]
```

Comment:
ActiveMerchant conventions specific that any values coming through options should be in snake case, so this should be changed to `post[:internalReference] = options[:internal_reference]`